File: CivitasRuby/pkg/CivitasRuby-0.0.1/lib/controlador.rb
#encoding:utf-8

require_relative 'vista_textual'
require_relative 'civitas_juego'
require_relative 'diario'
require_relative 'respuestas'
require_relative 'controlador'
require_relative 'operacion_inmobiliaria'
require_relative 'gestiones_inmobiliarias'
require_relative 'casilla'

module Civitas
  class Controlador
    

    def initialize(j,v)
      @juego=j
      @vista=v
    end
    
    def juega
      @vista.set_civitas_juego(@juego)
      casillamost=false
      
      while !@juego.final_del_juego

        @vista.pausa
       
        op= @juego.siguiente_paso

        @vista.mostrar_siguiente_operacion(op)
        
        if op != Operaciones_juego::PASAR_TURNO
          puts "\n*********************EVENTOS**********************\n"
          @vista.mostrar_eventos
          puts "**************************************************\n"
        else
          casillamost=false
        end
        
        if !casillamost
          @vista.actualizar_vista
          casillamost=true
        end
        
        
        if !@juego.final_del_juego
          case op
          when Operaciones_juego::COMPRAR
            if(@juego.get_jugador_actual.puede_comprar_casilla)
              res = @vista.comprar
              if res == Civitas::Respuestas::Lista_respuestas[1]
                @juego.comprar
                @juego.siguiente_paso_completado(op)
              elsif res == Civitas::Respuestas::Lista_respuestas[0]
                @juego.siguiente_paso_completado(op)
              end
            else
              @juego.siguiente_paso_completado(op)
            end
          when Operaciones_juego::GESTIONAR
            @vista.gestionar
            gest = @vista.get_gestion
            prop = @vista.get_propiedad
            op_inm = Operacion_inmobiliaria.new(Civitas::Gestiones_inmobiliarias::Lista_gestiones_inmobiliarias[gest],prop)
            case op_inm.get_gestion
            when Civitas::Gestiones_inmobiliarias::Lista_gestiones_inmobiliarias[0]
              @juego.vender(prop)
            when Civitas::Gestiones_inmobiliarias::Lista_gestiones_inmobiliarias[1]
              @juego.hipotecar(prop)
            when Civitas::Gestiones_inmobiliarias::Lista_gestiones_inmobiliarias[2]
              @juego.cancelar_hipoteca(prop)
            when Civitas::Gestiones_inmobiliarias::Lista_gestiones_inmobiliarias[3]
              @juego.construir_casa(prop)
            when Civitas::Gestiones_inmobiliarias::Lista_gestiones_inmobiliarias[4]
              @juego.construir_hotel(prop)
            when Civitas::Gestiones_inmobiliarias::Lista_gestiones_inmobiliarias[5]
              @juego.siguiente_paso_completado(op)
            end
          when Operaciones_juego::SALIR_CARCEL
            salidas = @vista.salir_carcel
            if salidas == Salidas_carcel::Lista_salidas_carcel[0]
              @juego.salir_carcel_pagando
            else
              @juego.salir_carcel_tirando
            end
            @juego.siguiente_paso_completado(op)
          end
        end
        
        @juego.info_jugador_texto
      end
      
      
      
      jugadores = @juego.ranking
      
      for i in jugadores do
        puts i
      end
    end
  end
end

File: CivitasJava/src/GUI/Controlador.java
package GUI;

import civitas.CivitasJuego;
import civitas.OperacionesJuego;
import civitas.OperacionInmobiliaria;

public class Controlador {
    private CivitasJuego juego;
    private CivitasView vista;
    
    public Controlador(CivitasJuego juego, CivitasView vista){
        this.juego=juego;
        this.vista=vista;
    }
    
    public void juega(){
        vista.setCivitasJuego(juego);

        boolean finalJuego=juego.finalDelJuego();
        boolean casillamost =false;

        OperacionesJuego op;

        while (!finalJuego){

            //vista.pausa();

            op=juego.siguientePaso();

            vista.mostrarSiguienteOperacion(op);

            if(op != OperacionesJuego.PASAR_TURNO){
                vista.mostrarEventos();
            }
            else{
                casillamost= false;
            }

            if(!casillamost){
                vista.actualizarVista();
                casillamost=true;
            }

            finalJuego=juego.finalDelJuego();

            
            if (!finalJuego){
                if(op==civitas.OperacionesJuego.COMPRAR){
                    if(juego.getJugadorActual().puedeComprarCasilla()){
                        GUI.Respuestas res = vista.comprar();
                        if(res == GUI.Respuestas.SI){
                            juego.comprar();
                            juego.siguientePasoCompletado(op);
                            vista.actualizarVista();
                        }
                        else if(res == GUI.Respuestas.NO){
                            juego.siguientePasoCompletado(op);
                        }
                    }
                    else{
                        juego.siguientePasoCompletado(op);
                    }
                }
                else if(op==civitas.OperacionesJuego.GESTIONAR){
                    vista.gestionar();
                    int gest = vista.getGestion();
                    int prop = vista.getPropiedad();
                    civitas.OperacionInmobiliaria op_inm = new OperacionInmobiliaria(civitas.GestionesInmobiliarias.values()[gest], prop);
                    if (op_inm.getGestion() == civitas.GestionesInmobiliarias.values()[0]){
                        juego.vender(prop);
                    }
                    else if (op_inm.getGestion() == civitas.GestionesInmobiliarias.values()[1]){
                        juego.hipotecar(prop);
                    }
                    else if (op_inm.getGestion() == civitas.GestionesInmobiliarias.values()[2]){
                        juego.cancelarHipoteca(prop);
                    }
                    else if (op_inm.getGestion() == civitas.GestionesInmobiliarias.values()[3]){
                        juego.construirCasa(prop);
                    }
                    else if (op_inm.getGestion() == civitas.GestionesInmobiliarias.values()[4]){
                        juego.construirHotel(prop);
                    }
                    else if (op_inm.getGestion() == civitas.GestionesInmobiliarias.values()[5]){
                        juego.siguientePasoCompletado(op);    
                    }
                }
                else if (op==civitas.OperacionesJuego.SALIR_CARCEL){
                    civitas.SalidasCarcel salidas = vista.salirCarcel();
                    if (salidas == civitas.SalidasCarcel.values()[0]){
                        juego.salirCarcelPagando();
                    }
                    else{
                        juego.salirCarcelTirando();
                    }
                    juego.siguientePasoCompletado(op);
                }
            }
            
            
        }
        
        vista.ranking();
    }
    
    
}
